core/src/main/java/com/github/mizool/core/concurrent/ListenableFutureCollector.java

```diff
@@ -150,15 +150,7 @@ private void waitWhileCapacityReached()
     {
         while (runningFutures.get() >= maximumConcurrentFutures)
         {
-            try
-            {
-                semaphore.wait();
-            }
-            catch (InterruptedException e)
-            {
-                Thread.currentThread().interrupt();
-                throw new UncheckedInterruptedException(e);
-            }
+            Threads.wait(semaphore);
```

Comment:
also use it on `BlockingAsynchronousTaskController` - no reason to leave that code ugly just because it's deprecated